src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeploymentPosterImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.atlassian.bitbucket.jenkins.internal.deployments;
+
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactory;
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactoryProvider;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.AuthorizationException;
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.GlobalCredentialsProvider;
+import com.atlassian.bitbucket.jenkins.internal.credentials.JenkinsToBitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketCDCapabilities;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import com.cloudbees.plugins.credentials.Credentials;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Singleton
+public class DeploymentPosterImpl implements DeploymentPoster {
+
+    protected static final Logger LOGGER = Logger.getLogger(DeploymentPosterImpl.class.getName());
+
+    @Inject
```

Comment:
I must say I would _very_ much prefer constructor injection instead. Makes it easier to breakpoint, log (during local development), and test (which is another thing there is no test for this class and there is enough logic below that I think it warrants a test)

Reply:
I said in the PR description (and mentioned it in stand-up this morning) that the unit tests are coming. I opened the PR for early feedback.

I don't understand why it makes it easier to test though, since `@InjectMocks` injects them just fine without it

Reply:
>I said in the PR description (and mentioned it in stand-up this morning) that the unit tests are coming.

You totally did, I forgot it when I was here, sorry

>but I don't understand why a constructor makes it easier to test though since @InjectMocks injects them just fine without it

You are right, but you must now always either use the CI framework or `@injectMocs` to get hold of an instance. If you want to inject one custom class and the rest mocks, you have to use a `@spy`. If you want to quickly test something (not nervelessly a unit test, but a manual test) it is easier with constructor injection. Plus I see no real upside to having property injection, granted you don't have the constructor but that is not a huge benefit.

Reply:
This still holds, I still very much prefer constructor injection over property injection.